core/trino-main/src/main/java/io/trino/sql/planner/optimizations/StreamPreferredProperties.java

```diff
@@ -105,6 +105,16 @@ public StreamPreferredProperties withFixedParallelism()
         return fixedParallelism();
     }
 
+    public static StreamPreferredProperties partitionedOn(Collection<Symbol> partitionSymbols)
+    {
+        if (partitionSymbols.isEmpty()) {
+            return singleStream();
+        }
+
+        // this must be the partitioning symbols, in any order
```

Comment:
i don't understand the comment (although is see it's copied from `exactlyPartitionedOn` and adapted)